fiftyone/server/routes/tagging.py

```
@@ -53,37 +62,258 @@ async def post(self, request: Request, data: dict) -> dict:
+            dataset,
+        )
+
+        if is_filtered:
+            all_tags = await _get_all_dataset_tags(dataset, target_labels)
+            t2 = time.perf_counter()
+            logger.warning(
+                "[tagging] all_tags (dataset scan) took %.3fs",
+                t2 - t1,
+            )
+        else:
+            all_tags = sorted(t for t in tags.keys() if t is not None)
```

Ruff (0.15.9)

[warning] Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)